### PR TITLE
Base HLSL URP Shader

### PR DESCRIPTION
**Pull request by toryjleo**

### CODE REVIEW

File: Assets/Code/Shaders/MyLit/MyLit.shader [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+
+
+#if UNITY_VERSION >= 202120
+            #pragma multi_compile _ _MAIN_LIGHT_SHADOWS _MAIN_LIGHT_SHADOWS_CASCADE
+#else
+            #pragma multi_compile _ _MAIN_LIGHT_SHADOWS // Tells unity to compile a version of the shader that works with and without main light shadows. This creates variants of the forward lit pass
+            #pragma multi_compile _ _MAIN_LIGHT_SHADOWS_CASCADE // Handles soft shadows automatically
+#endif
+            #pragma multi_compile_fragment _ _SHADOWS_SOFT      // Handles soft shadows automatically
+#if UNITY_VERSION >= 202120
```
Comment (LiamMillerSE) [inline Assets/Code/Shaders/MyLit/MyLit.shader:84]: Does this need to be a second if statement

File: Assets/Scripts/Editor/MyLitCustomInpector.cs [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        {
+            UpdateSurfaceType(material);
+        }
+
+        base.OnGUI(materialEditor, properties);
+    }
+
+    private void UpdateSurfaceType(Material material)
+    {
+        //--- SurfaceType ---TODO: Make Method
```
Comment (LiamMillerSE) [inline Assets/Scripts/Editor/MyLitCustomInpector.cs:54]: Unresolved TODO: comment ?

File: Assets/Scripts/Editor/MyLitCustomInpector.cs [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+        if (surface == SurfaceType.TransparentCutout) 
+        {
+            material.EnableKeyword("_ALPHA_CUTOUT");
+        }
+        else 
+        {
+            material.DisableKeyword("_ALPHA_CUTOUT");
+        }
+
+        //--- FaceRenderingMode ---TODO: Make Method
```
Comment (LiamMillerSE) [inline Assets/Scripts/Editor/MyLitCustomInpector.cs:98]: Unresolved TODO: comment ?

File: Assets/Scripts/Editor/MyLitCustomInpector.cs [hunk 4]
```diff
@@ -0,0 +1,161 @@
...
+        if (faceRenderingMode == FaceRenderingMode.DoubleSided)
+        {
+            material.EnableKeyword("_DOUBLE_SIDED_NORMALS");
+        }
+        else
+        {
+            material.DisableKeyword("_DOUBLE_SIDED_NORMALS");
+        }
+
+        // --- BlendType ---TODO: Make Method
```
Comment (LiamMillerSE) [inline Assets/Scripts/Editor/MyLitCustomInpector.cs:118]: TODO comment

**Review by LiamMillerSE — APPROVED** [review]
Some things to take a look at but otherwise looks good

File: Assets/Scripts/Editor/MyLitCustomInpector.cs [hunk 5]
```diff
@@ -0,0 +1,161 @@
+using UnityEditor;
+using UnityEngine;
+using UnityEngine.Rendering;
+
```
Comment (toryjleo) [inline Assets/Scripts/Editor/MyLitCustomInpector.cs:4]: Add Class description

File: Assets/Code/Shaders/MyLit/MyLit.shader (hunk 1 above)
Comment (toryjleo) [reply Assets/Code/Shaders/MyLit/MyLit.shader:84]: Nope!

**Review by LiamMillerSE — APPROVED** [review]

**Pull request merged**